packages/aws-cdk-lib/aws-appsync/lib/graphqlapi.ts

```diff
@@ -702,32 +695,44 @@ export class GraphqlApi extends GraphqlApiBase {
   }
 
   private setupSourceApiAssociations() {
-    this.definition.sourceApiOptions?.sourceApis.forEach(sourceApiOption => {
-      new CfnSourceApiAssociation(this, `${sourceApiOption.sourceApi.node.id}Association`, {
-        sourceApiIdentifier: sourceApiOption.sourceApi.apiId,
-        mergedApiIdentifier: this.api.attrApiId,
+    this.definition.sourceApiOptions?.sourceApis.forEach(sourceApiConfig => {
+      const mergeType = sourceApiConfig.mergeType ?? MergeType.AUTO_MERGE;
+      var sourceApiIdentifier = sourceApiConfig.sourceApi.apiId;
+      var mergedApiIdentifier = this.apiId;
+
+      if (FeatureFlags.of(this).isEnabled(cxapi.APPSYNC_ENABLE_USE_ARN_IDENTIFIER_SOURCE_API_ASSOCIATION)) {
```

Comment:
Can we just add a comment here talking about why this is behind a feature flag, IE resource replacement even though the service supports both IDs and ARNs, only ARNs being supported cross region and changes resulting in resource replacement etc.

Reply:
Sure